cloudinit/net/__init__.py

```diff
@@ -531,12 +535,14 @@ def find_candidate_nics_on_linux() -> List[str]:
     # 2. Remaining connected interfaces, naturally sorted.
     # 3. DEFAULT_PRIMARY_INTERFACE, if possibly connected.
     # 4. Remaining possibly connected interfaces, naturally sorted.
+    if not default_primary_interface:
+        default_primary_interface = DEFAULT_PRIMARY_INTERFACE
```

Comment:
Is it worth a breadcrumb comment referencing # provided by DataSourceGCE?

Reply:
Added.